File: Day14/part1.py
with open("./Day14/input.txt") as f:
    data = [line.strip() for line in f.readlines()]

cols = [[line[colNum] for line in data] for colNum in range(len(data[0]))]

count = 0

for col in cols:
    lastStop = -1
    for i in range(len(cols)):
        if col[i] == "O":
            col.insert(lastStop + 1, col.pop(i))
        elif col[i] == "#":
            lastStop = i

for col in cols:
    for i, char in enumerate(col):
        if char == "O":
            count += len(col) - i

print(count)

File: Day14/part1.ts
import fs from "fs";

const data = fs.readFileSync("./input.txt", "utf-8").split("\n");

const cols = data[0]
  .split("")
  .map((_, colNum) => data.map((line) => line[colNum]));
console.log(cols);

for (let col of cols) {
  let lastStop = -1;
  for (let i = 0; i < col.length; i++) {
    if (col[i] === "O") {
      col.i;
    }
  }
}
